tsconfig.json

```diff
@@ -2,6 +2,8 @@
   "include": ["src"],
   "compilerOptions": {
     "lib": ["es2015"],
+    // TODO: change to ES5 and make sure maps still work
```

Comment:
I tried to test current implementation with extending map (I like it and I'm curious how it works :) 
I found some issue with TS output to ES5 and V8 
```
Uncaught TypeError: Constructor Map requires 'new'
    at DraftMap.Map (<anonymous>)
    at new DraftMap (<anonymous>:19:28)
    at <anonymous>:1:1
```
https://github.com/microsoft/TypeScript/issues/10853

Reply:
Yeah, that is how far I got :D. Probably, if extending is not possible, we should do a dirty set prototype hack. The most important thing is that `draftMap instanceof Map` passes, beyond that we don't really make / need any super calls

Reply:
This is probably an interesting thing to figure out if you are interested :) Beyond that it might be a bit hard to help, as I am a bit all over the place, refactoring to reduce code duplication, (or just untangle some spaghetti that grew over time), so it is kinda hard to split that up